TST/BUG: chisquare test with new scipy raises exception, tail observations are ignored in unit test

test failure with future scipy
https://dev.azure.com/statsmodels/statsmodels-testing/_build/results?buildId=3091&view=logs&j=d3aeec9b-5735-51da-6bfa-db4d86dad819&t=4a8eef27-4d92-5b93-41e2-3d1348afd29c

observed and predicted don't add up to 1 closely enough

```
self = <statsmodels.discrete.tests.test_discrete.TestGeneralizedPoisson_underdispersion object at 0x7f25270c6d30>

    def test_predict_prob(self):
        res = self.res
        endog = res.model.endog
        freq = np.bincount(endog.astype(int))
    
        pr = res.predict(which='prob')
        pr2 = sm.distributions.genpoisson_p.pmf(np.arange(6)[:, None],
                                        res.predict(), res.params[-1], 1).T
        assert_allclose(pr, pr2, rtol=1e-10, atol=1e-10)
    
        from scipy import stats
>       chi2 = stats.chisquare(freq, pr.sum(0))
```

failure is exception raised now by scipy

```
>               raise ValueError(msg)
E               ValueError: For each axis slice, the sum of the observed frequencies must agree with the sum of the expected frequencies to a relative tolerance of 1e-08, but the percent differences are:
E               0.0014010692380163618
```

I just saw that there is a second test failure in `pre` release environment

scipy.stats.skew raises

```
        m3 = moment(a, 3, axis)
        with np.errstate(all='ignore'):
>           zero = (m2 <= (np.finfo(m2.dtype).resolution * a.mean(axis))**2)
E           AttributeError: 'float' object has no attribute 'dtype
```

scipy skew is called from our jarque_bera, which is called from stats.descriptivestats

```
statsmodels/stats/descriptivestats.py:450: in <lambda>
    lambda x: list(jarque_bera(x.dropna())), result_type="expand"
statsmodels/stats/stattools.py:123: in jarque_bera
    skew = stats.skew(resids, axis=axis
```

I don't see anything wrong, mabye/likely a bug in scipy.
jarque_bera uses np.asarray, so pandas shouldn't be involved when calling skew.

--- a/statsmodels/discrete/tests/test_discrete.py
+++ b/statsmodels/discrete/tests/test_discrete.py
@@ -1866,9 +1866,16 @@ class TestGeneralizedPoisson_underdispersion(object):
                                         res.predict(), res.params[-1], 1).T
         assert_allclose(pr, pr2, rtol=1e-10, atol=1e-10)
 
+        expected = pr.sum(0)
+        # add expected obs from right tail to last bin
+        expected[-1] += pr.shape[0] - expected.sum()
+        # scipy requires observed and expected add to the same at rtol=1e-8
+        assert_allclose(freq.sum(), expected.sum(), rtol=1e-13)
+
         from scipy import stats
-        chi2 = stats.chisquare(freq, pr.sum(0))
-        assert_allclose(chi2[:], (0.64628806058715882, 0.98578597726324468),
+        chi2 = stats.chisquare(freq, expected)
+        # numbers are regression test, we should not reject
+        assert_allclose(chi2[:], (0.5511787456691261, 0.9901293016678583),
                         rtol=0.01)
 
 

